### ISSUE TITLE
[Question] Possible to customize the color of the Mark Points?

### ISSUE THREAD

**Issue by flaneur7508** [go-echarts/go-echarts#456]
### Describe your question

I would like to change the markpoint color based on the value. i.e. for Max values i'd like the markpoint to be Green, for average values, Orange and Min values Red. 
Also if the value is between 100 and 1000 i'd like the markpoint to be purple. 

Is there a way to do this? 

### go-echarts version

v2.4.0-rc2

**Comment by Koooooo-7** [comment]
Hi @flaneur7508 .
For the `Min` `Max` and `Average` value, you can config like this.
```go
	line.SetXAxis(x).AddSeries("myLine", y).
		SetSeriesOptions(
			charts.WithMarkPointNameTypeItemOpts(
				opts.MarkPointNameTypeItem{Name: "Maximum", Type: "max", ItemStyle: &opts.ItemStyle{Color: "rgb(0, 128, 0)"}},
				opts.MarkPointNameTypeItem{Name: "Average", Type: "average", ItemStyle: &opts.ItemStyle{Color: "rgb(255, 165, 0)"}},
				opts.MarkPointNameTypeItem{Name: "Minimum", Type: "min", ItemStyle: &opts.ItemStyle{Color: "rgb(255, 0, 0)"}},
			)
```
> Also if the value is between 100 and 1000 i'd like the markpoint to be purple.

Unfortunately, for the `data` range, there is no such a simple config to do it.

**Comment by flaneur7508** [comment]
That's great. Thank you!
I tried to use the same model to change the color of the MarkLineNameTypeItem's on a BarChart but that does not seem to work because ItemStyle does not seem to be supported with MarkLineNameTypeItem's . What am I missing?

**Comment by Koooooo-7** [comment]
No prob.
This is no `ItemStyle` for the MarkLine, instead, it has the `LineStyle`.
```go
	bar.SetXAxis(x).AddSeries("foo", y).
		SetSeriesOptions(
			charts.WithMarkLineNameTypeItemOpts(opts.MarkLineNameTypeItem{
				Name: "Maximum",
				Type: "max",
			}),
			charts.WithMarkLineStyleOpts(opts.MarkLineStyle{LineStyle: &opts.LineStyle{Color: "rgb(0, 128, 0)"}}),
		)

```

**Comment by flaneur7508** [comment]
I seem to get an error with that:

	bar.SetXAxis(startMonthNames).AddSeries("order value", barDataOrderValue).
		SetSeriesOptions(
			charts.WithMarkLineNameTypeItemOpts(opts.MarkLineNameTypeItem{
				Name: "Maximum",
				Type: "max",
			}),
			charts.WithMarkLineStyleOpts(opts.MarkLineStyle{LineStyle: &opts.**LineStyle**{Color: "rgb(0, 128, 0)"}}),
		)

I get  Unknown field 'LineStyle' in struct literal.

**Comment by Koooooo-7** [comment]
Ooops, I see, the new added options is unreleased, plz upgrade to `v2.4.0` the standard release version.

**Comment by flaneur7508** [comment]
Yes, that compiles, thank you. Once final question if I may though, Is it possible to set the different lines to be different colors? My code below which compiles results in all lines being displayed in Blue.

`bar.SetXAxis(startMonthNames).
    AddSeries("Order value", barDataOrderValue).
    SetSeriesOptions(
        charts.WithMarkLineNameTypeItemOpts(
            opts.MarkLineNameTypeItem{Name: "Minimum", Type: "min"},
        ),
        charts.WithMarkLineStyleOpts(
            opts.MarkLineStyle{
                Label:     &opts.Label{FontSize: 15},
                LineStyle: &opts.LineStyle{Color: "red"},
            },
        ),
        charts.WithMarkLineNameTypeItemOpts(
            opts.MarkLineNameTypeItem{Name: "Maximum", Type: "max"},
        ),
        charts.WithMarkLineStyleOpts(
            opts.MarkLineStyle{
                Label:     &opts.Label{FontSize: 15},
                LineStyle: &opts.LineStyle{Color: "green"},
            },
        ),
        charts.WithMarkLineNameTypeItemOpts(
            opts.MarkLineNameTypeItem{Name: "Average", Type: "average"},
        ),
        charts.WithMarkLineStyleOpts(
            opts.MarkLineStyle{
                Label:     &opts.Label{FontSize: 15},
                LineStyle: &opts.LineStyle{Color: "blue"},
            },
        ),
    )`

**Comment by Koooooo-7** [comment]
Hi, I suppose the color doesn't support the name-color, have a try to use the `RGB` instead.

**Comment by flaneur7508** [comment]
Hi, unfortunately the same behaviour is found using rgb.

**Comment by Koooooo-7** [comment]
> Hi, unfortunately the same behaviour is found using rgb.
> 

Sounds weird. I will try to reproduce it asap.

**Comment by Koooooo-7** [comment]
It seems a missing option for setting each line style per mark line data item.
It only allows to config the global line style and last wins.

**Comment by Koooooo-7** [comment]
Hi @flaneur7508 , I have a patch on this, hence you can config like this when it does the next release:
```go
func barBasic() *charts.Bar {
	bar := charts.NewBar()
	bar.SetGlobalOptions(
		charts.WithTitleOpts(opts.Title{Title: "basic bar example", Subtitle: "This is the subtitle."}),
	)

	bar.SetXAxis(weeks).
		AddSeries("Category A", generateBarItems()).
		SetSeriesOptions(
			charts.WithMarkLineNameTypeItemOpts(
				opts.MarkLineNameTypeItem{Name: "Minimum", Type: "min", LineStyle: &opts.LineStyle{Color: "rgb(255, 0, 0)"}},
				opts.MarkLineNameTypeItem{Name: "Maximum", Type: "max", LineStyle: &opts.LineStyle{Color: "rgb(255, 165, 0)"}},
				opts.MarkLineNameTypeItem{Name: "Average", Type: "average", LineStyle: &opts.LineStyle{Color: "blue"}},
			))
	return bar

```
![mk](https://github.com/go-echarts/go-echarts/assets/33706142/908ee8f1-fc13-4195-9a7a-2bf184a2f290)

I suppose u could use the latest master commits version if you do need it in hurry.
Sorry for the inconvenient.

**Comment by flaneur7508** [comment]
Perfect, thank you!
